src/carma_clock.cpp

```diff
@@ -85,6 +86,10 @@ void CarmaClock::wait_for_initialization() {
     }
 }
 
+void CarmaClock::sleep_for(timeStampMilliseconds time_to_sleep) {
+    sleep_until( _current_time + time_to_sleep);
```

Comment:
In the sleep_until function, there is a wait call ```sleepCVPairValue.first->wait(lock); ``` How does that get notified?

Reply:
In the update function there is a mechanism that notifies each conditional variable based on whether the most recent update is larger or equal to the sleep until value.

Reply:
Can we added description about those two conditional variables  _initialization_cv and sleepCVPair. What they do and why they are needed?

Reply:
Added addition comments to header with descriptions of the variable used.